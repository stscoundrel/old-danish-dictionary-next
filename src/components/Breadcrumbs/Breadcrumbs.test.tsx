import ReactDOM from 'react-dom/client'
import renderer from 'react-test-renderer'
import Breadcrumbs from './index'

describe('Breadcrumbs component', () => {
  const word = {
    headword: 'GrÃ¸nfugl',
    definitions: [
      'no. totanus. Moth. Smlgn. grittue.',
    ],
    slug: 'gronfugl',
  }

  const letter = {
    letter: 'g',
    slug: 'g',
  }

  test('Default page: does not crach', () => {
    const div = document.createElement('div')
    const root = ReactDOM.createRoot(div)
    root.render(<Breadcrumbs word={null} letter={null} />)
  })

  test('Default page: matches snapshot', () => {
    const tree = renderer.create(<Breadcrumbs word={null} letter={null} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Default page: renders correct amount of breadcrumbs', () => {
    const tree = renderer.create(<Breadcrumbs word={null} letter={null} />)
    const { root } = tree

    expect(root.findAllByType('a').length).toEqual(1)
  })

  test('Letter page: does not crach', () => {
    const div = document.createElement('div')
    const root = ReactDOM.createRoot(div)
    root.render(<Breadcrumbs word={null} letter={letter} />)
  })

  test('Letter page: matches snapshot', () => {
    const tree = renderer.create(
      <Breadcrumbs word={null} letter={letter} />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Letter page: renders correct amount of breadcrumbs', () => {
    const tree = renderer.create(<Breadcrumbs word={null} letter={letter} />)
    const { root } = tree

    expect(root.findAllByType('a').length).toEqual(2)
  })

  test('Word page: does not crach', () => {
    const div = document.createElement('div')
    const root = ReactDOM.createRoot(div)
    root.render(<Breadcrumbs word={word} letter={letter} />)
  })

  test('Word page: matches snapshot', () => {
    const tree = renderer.create(<Breadcrumbs word={word} letter={letter} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Word page: renders correct amount of breadcrumbs', () => {
    const tree = renderer.create(<Breadcrumbs word={word} letter={letter} />)
    const { root } = tree

    expect(root.findAllByType('a').length).toEqual(3)
  })
})
