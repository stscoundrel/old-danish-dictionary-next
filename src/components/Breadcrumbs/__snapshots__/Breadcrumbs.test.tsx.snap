// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs component Default page: matches snapshot 1`] = `
<nav
  className="section"
>
  <div
    className="container"
  >
    <a
      className="link"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Old Danish Dictionary
    </a>
    <script
      dangerouslySetInnerHTML={
        {
          "__html": "{"@context":"https://schema.org/","@type":"BreadcrumbList","itemListElement":[{"@type":"ListItem","position":1,"name":"Old Danish Dictionary","item":"undefined/"}]}",
        }
      }
      type="application/ld+json"
    />
  </div>
</nav>
`;

exports[`Breadcrumbs component Letter page: matches snapshot 1`] = `
<nav
  className="section"
>
  <div
    className="container"
  >
    <a
      className="link"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Old Danish Dictionary
    </a>
    <a
      className="link"
      href="/letter/g"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Letter G
    </a>
    <script
      dangerouslySetInnerHTML={
        {
          "__html": "{"@context":"https://schema.org/","@type":"BreadcrumbList","itemListElement":[{"@type":"ListItem","position":1,"name":"Old Danish Dictionary","item":"undefined/"},{"@type":"ListItem","position":2,"name":"Letter G","item":"undefined/letter/g"}]}",
        }
      }
      type="application/ld+json"
    />
  </div>
</nav>
`;

exports[`Breadcrumbs component Word page: matches snapshot 1`] = `
<nav
  className="section"
>
  <div
    className="container"
  >
    <a
      className="link"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Old Danish Dictionary
    </a>
    <a
      className="link"
      href="/letter/g"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Letter G
    </a>
    <a
      className="link"
      href="/word/gronfugl"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Grønfugl
    </a>
    <script
      dangerouslySetInnerHTML={
        {
          "__html": "{"@context":"https://schema.org/","@type":"BreadcrumbList","itemListElement":[{"@type":"ListItem","position":1,"name":"Old Danish Dictionary","item":"undefined/"},{"@type":"ListItem","position":2,"name":"Letter G","item":"undefined/letter/g"},{"@type":"ListItem","position":3,"name":"Grønfugl","item":"undefined/word/gronfugl"}]}",
        }
      }
      type="application/ld+json"
    />
  </div>
</nav>
`;
